specs/applicationdata/ApplicationData.md

```diff
@@ -0,0 +1,428 @@
+        Local = 0,
+        LocalCache = 3,
+        Roaming = 1,
+        SharedLocal = 4,
+        Temporary = 2,
+        Machine = 1000,
+    };
+
+    /// Specifies options for creating application data containers or returning existing containers.
+    [contract(ApplicationDataContract, 1)]
+    enum ApplicationDataCreateDisposition
+    {
+        /// Always returns the specified container. Creates the container if it does not exist.
+        Always = 0,
+
+        // Returns the specified container only if it already exists. Raises an exception of type System.Exception if the specified container does not exist.
+        Existing,
+    };
+
+    /// Represents a container for settings. The methods and properties of this class support creating, deleting, enumerating, and traversing the container hierarchy.
+    /// @see https://learn.microsoft.com/uwp/api/windows.storage.applicationdatacontainer
+    [contract(ApplicationDataContract, 1)]
+    runtimeclass ApplicationDataContainer : Windows.Foundation.IClosable
+    {
+        /// Return the child settings containers of the settings container.
+        /// https://learn.microsoft.com/uwp/api/windows.storage.applicationdatacontainer.containers
+        IMap<String, ApplicationDataContainer> Containers { get; };
+
+        /// Return the name of the current settings container.
+        /// https://learn.microsoft.com/uwp/api/windows.storage.applicationdatacontainer.name
+        String Name { get; };
+
+        /// Return the type of the app data store that is associated with the settings container.
+        /// @see https://learn.microsoft.com/uwp/api/windows.storage.applicationdatacontainer.locality
+        ApplicationDataLocality Locality { get; };
+
+        /// Return an object that represents the settings in the settings container.
+        /// @see https://learn.microsoft.com/uwp/api/windows.storage.applicationdatacontainer.values
+        Windows.Foundation.Collections.IPropertySet Values { get; };
+
+        /// Create or open the specified settings container in the settings container.
+        /// @see https://learn.microsoft.com/uwp/api/windows.storage.applicationdatacontainer.createcontainer
+        ApplicationDataContainer CreateContainer(String name, ApplicationDataCreateDisposition disposition);
+
+        /// Delete the specified settings container, its subcontainers, and all settings in the hierarchy.
+        /// @see https://learn.microsoft.com/uwp/api/windows.storage.applicationdatacontainer.deletecontainer
+        void DeleteContainer(String name);
+    };
+
+    /// Provides access to the application data store.
+    /// @see https://learn.microsoft.com/uwp/api/windows.storage.applicationdata
+    [contract(ApplicationDataContract, 1)]
+    runtimeclass ApplicationData : Windows.Foundation.IClosable
+    {
+        /// Get an instance of ApplicationData for the current user.
+        /// @note Requires the current process has package identity.
```

Comment:
What happens when called from unpackaged? An exception?

Reply:
Yes. Same as an unpackaged process calling `Windows.Storage.ApplicationData.Current`, you get HRESULT_FROM_WIN32(APPMODEL_ERROR_NO_PACKAGE).

Unless we come up with a way for an unpackaged app to implicitly provide the information (e.g. via embedded resource, as discussed in other comment).